### PR TITLE
Inav f7 common changes

### PR DESCRIPTION
**Pull request by sambas**
Inav f7 common changes

### CODE REVIEW

**Comment by digitalentity** [comment]
OMG! Why this has to be so huge!!!

**Comment by martinbudden** [comment]
@sambas , can you rebase this now that the F7 libs have been merged in.

@digitalentity - it should be quite a bit smaller after rebase.

File: src/main/build/atomic.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ __attribute__( ( always_inline ) ) static inline void __set_BASEPRI_MAX_nb(uint3
    __ASM volatile ("\tMSR basepri_max, %0\n" : : "r" (basePri) );
 }
 
-#ifndef STM32F4 /* already defined in /lib/main/CMSIS/CM4/CoreSupport/core_cmFunc.h for F4 targets */
+#if !defined(STM32F4) && !defined(STM32F7) /* already defined in /lib/main/CMSIS/CM4/CoreSupport/core_cmFunc.h for F4 targets */
```
Comment (digitalentity) [inline src/main/build/atomic.h]: With upgraded CMSIS this is no longer necessary

File: src/main/config/config_eeprom.c [hunk 2]
```diff
@@ -168,6 +170,70 @@ static uint32_t getFLASHSectorForEEPROM(void)
...
+    masterConfig.magic_ef = 0xEF;
+    masterConfig.chk = 0; // erase checksum before recalculating
+    masterConfig.chk = calculateChecksum((const uint8_t *) &masterConfig, sizeof(master_t));
+
+    // write it
+    /* Unlock the Flash to enable the flash control register access *************/
+    HAL_FLASH_Unlock();
+    while (attemptsRemaining--)
+    {
+        /* Fill EraseInit structure*/
```
Comment (digitalentity) [inline src/main/config/config_eeprom.c:200]: This has to be changed to respect the linker script and put configuration into FLASH_CONFIG

File: src/main/drivers/bus_i2c_hal.c [hunk 3]
```diff
@@ -0,0 +1,324 @@
+/*
```
Comment (digitalentity) [inline src/main/drivers/bus_i2c_hal.c:1]: I2C driver will have to be refactored to the same state-machine structure as current F1/F3/F4 i2c drivers. But this is a lower priority

File: src/main/drivers/pwm_output_hal.c [hunk 4]
```diff
@@ -0,0 +1,245 @@
+/*
```
Comment (digitalentity) [inline src/main/drivers/pwm_output_hal.c:1]: Can't this file be unified with current implementation of `pwm_output.c`? Most of the code stays the same anyway and it's a pain to maintain two almost identical files.

**Pull request merged**